module/jobqueue/component_consumer_test.go

```diff
@@ -125,17 +125,19 @@ func (suite *ComponentConsumerSuite) TestHappyPath() {
 	finishedJobs := make(map[uint64]bool, testJobsCount)
 
 	wg := sync.WaitGroup{}
+	mu := sync.Mutex{}
 
 	processor := func(_ irrecoverable.SignalerContext, _ module.Job, complete func()) { complete() }
 	notifier := func(jobID module.JobID) {
 		defer wg.Done()
 
 		index, err := JobIDToIndex(jobID)
 		assert.NoError(suite.T(), err)
+		suite.T().Logf("job %d finished", index)
 
+		mu.Lock()
+		defer mu.Unlock()
 		finishedJobs[index] = true
-
-		suite.T().Logf("job %d finished", index)
```

Comment:
why move the message that job is finished to before the `finishedJobs` map is updated?